scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2109,55 @@ def visit_call(self, node):
                     break
 
 
+class NoBlankLineBelowFunctionDefinition(checkers.BaseChecker):
+    """Checks if there are no blank lines available below function
+    definition.
+    """
+
+    _implements_ = interfaces.IAstroidChecker
+    name = 'newline-below-function-definition'
+    priority = -1
+    msgs = {
+        'W0034': (
+            'Please remove the empty line below the function definition.',
+            'empty-line-provided-below-function-definiton',
+            'Disallow blank lines below function definition.'
+        ),
+        'W0035': (
+            'No blank space below function definition.',
+            'no-empty-line-provided',
+            'No empty line should be provided..'
+        ),
+    }
+
+    def check_blank_lines_below_function_def(self, node):
+        """Visit each function and checks if there are no
+        new lines below function definition.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node to access module content.
+        """
+        # Check if the given node has docstring.
+        if node.doc is None:
```

Comment:
Why are you skipping the check if the function has no docstring? It is still possible that there are blank lines below function definition, right?

Reply:
I  skipped because it's wrong for a function to be without docstring according to the coding guidelines.But my checker is to test for blank lines and the function should only focus on that part, so I get your point. Hence I have removed this skip and corrected this.

Reply:
> I skipped because it's wrong for a function to be without docstring according to the coding guidelines.

I don't think that's true, a quick exception are methods written in test files (you also wrote some in this PR!). But we would still want to enforce this check for those functions. So it is not a very general assumption.